### PR TITLE
Using MFEM hosted dependencies in CI context

### PR DESCRIPTION
**Pull request by adrienbernede**
A new repo was added to MFEM org with the TPLs tarballs.

This PR let Travis and AppVeyor point to these.

Resolves https://github.com/mfem/mfem/issues/1631.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1635](https://github.com/mfem/mfem/pull/1635) | @adrienbernede | @tzanio | @tzanio + @v-dobrev | 07/17/20 | 07/20/20 | 07/20/20 | |

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ install:
 - set PATH=C:\Program Files\Microsoft MPI\Bin;%PATH%
 
 # Install METIS
-- ps: Start-FileDownload 'http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/metis-5.1.0.tar.gz'
+- ps: Start-FileDownload 'https://github.com/mfem/tpls/raw/gh-pages/metis-5.1.0.tar.gz'
```
Comment (tzanio) [inline .appveyor.yml:19]:
```suggestion
- ps: Start-FileDownload 'https://mfem.github.io/tpls/metis-5.1.0.tar.gz'
```

File: .travis.yml [hunk 2]
```diff
@@ -354,7 +354,7 @@ install:
    # METIS
    - if [ $MPI == "YES" ]; then
         if [ ! -e metis-4.0/libmetis.a ]; then
-           wget http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/OLD/metis-4.0.3.tar.gz;
+           wget https://github.com/mfem/tpls/raw/gh-pages/metis-4.0.3.tar.gz;
```
Comment (tzanio) [inline .travis.yml:357]:
```suggestion
           wget https://mfem.github.io/tpls/metis-4.0.3.tar.gz;
```

File: .appveyor.yml [hunk 3]
```diff
@@ -26,17 +26,17 @@ install:
 - cd ..
 
 # Install hypre
-- ps: Start-FileDownload 'https://github.com/hypre-space/hypre/archive/V2-10-0b.tar.gz'
-- 7z x V2-10-0b.tar.gz -so | 7z x -si -ttar > nul
-- cd hypre-2-10-0b
+- ps: Start-FileDownload 'https://github.com/mfem/tpls/raw/gh-pages/hypre-2.10.0b.tgz'
```
Comment (tzanio) [inline .appveyor.yml:29]: Keep the previous version, i.e use the GitHub archive in `hypre-space/hypre`

File: .travis.yml [hunk 4]
```diff
@@ -336,7 +336,7 @@ install:
    # hypre
    - if [ $MPI == "YES" ]; then
         if [ ! -e hypre-2.10.0b/src/hypre/lib/libHYPRE.a ]; then
-           wget https://computation.llnl.gov/project/linear_solvers/download/hypre-2.10.0b.tar.gz --no-check-certificate;
+           wget https://github.com/mfem/tpls/raw/gh-pages/hypre-2.10.0b.tar.gz --no-check-certificate;
```
Comment (tzanio) [inline .travis.yml:339]:
```suggestion
           wget https://github.com/hypre-space/hypre/archive/V2-10-0b.tar.gz --no-check-certificate;
```

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by adrienbernede** [comment]
LGTM

**Pull request merged**